File: StackQueue/기능개발.java
import java.util.ArrayList;
import java.util.LinkedList;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] days = new int[progresses.length];
        
        for(int i=0; i<progresses.length; i++){
            days[i] = (99 - progresses[i]) / speeds[i] + 1;
        }
        
        ArrayList<Integer> list = new ArrayList<>();
        LinkedList<Integer> q = new LinkedList<>();
        
        
        for(int i=0; i<days.length; i++){
            if(!q.isEmpty() && days[i] > q.peek()) {
                list.add(q.size());
                q.clear();
            }
            q.offer(days[i]);
        }
        list.add(q.size());
        q.clear();
        
        int size = list.size();
        int[] answer = new int[size];
        for(int i=0; i<size; i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
}

File: stackqueue/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int n = progresses.size();
    vector<int> days(n);
    for(int i = 0; i < n; i++) {
        days[i] = (100 - progresses[i]) / speeds[i];
        if((100 - progresses[i]) % speeds[i] != 0)
            days[i]++;
    }
    
    int cnt = 1, day = days[0];
    for(int i = 1; i < n; i++) {
        if(day < days[i]) {
            answer.push_back(cnt);
            cnt = 1;
            day = days[i];
        }
        else {
            cnt++;
        }
    }
    
    answer.push_back(cnt);
    
    return answer;
}